src/classes/PMT_Payment_TDTM.cls

```diff
@@ -132,6 +159,33 @@ public class PMT_Payment_TDTM extends TDTM_Runnable {
         // the Opportunity should still be marked as closed.
         PMT_PaymentCreator pc = new PMT_PaymentCreator(mapIdOppNew, mapIdOppOld, triggerAction, dmlWrapper);
 
+        //Prevent user modify Opp Amount
+        if (triggerAction == TDTM_Runnable.Action.BeforeUpdate && isEnforceAccountingDataConsistency) {
+            Set<Id> oppIds = (new Map<Id, Opportunity>(newList)).keySet();
+
+            Map<Id, Opportunity> oppsById = getPaymentOpportunities(oppIds);
+
+            for (Opportunity opp : oppsById.values()) {
+                Decimal totalPaid = 0;
+
+                for (npe01__OppPayment__c pmt : opp.npe01__OppPayment__r) {
+                    if (pmt.npe01__Paid__c) {
+                        totalPaid += pmt.npe01__Payment_Amount__c == null ? 0 : pmt.npe01__Payment_Amount__c;
```

Comment:
why not make this 

totalPaid += (pmt.npe01__Paid__c && pmt.npe01__Payment_Amount__c != null) ? pmt.npe01__Payment_Amount__c : 0;

This eliminates the if, and puts are affirmative choice first.